### ISSUE TITLE
Implement algorithm for Line Search, on a toy example to tweak the learning rate.

### ISSUE THREAD

**Issue by jorgemunozl** [jorgemunozl/geminiAPImemory#1]
Why they called it Line Search? I think that it's because you literally compared two lines.

Now most important, here we are not dealing with Neural Network but just with classical function that we want to minimize, you go from first principles.

Say you want to minimize a function, you have different methods, there is the newton, line search, or any variant of a steepest descent. You don't use a Neural Network.
